### PR TITLE
refactor builtin-pipeline

### PR DESCRIPTION
**Pull request by star-e**
Fix builtin-pipeline post-processing rendering bug when using multiple cameras.
Refactor builtin-pipeline and add builtin effects. Previous post-process effects are deprecated and will be removed in the future.

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

File: cc.config.json [hunk 1]
```diff
@@ -205,7 +205,7 @@
                 "84ac6f69-3086-455a-86a4-561da8ee710b",
                 "5c601d96-e4c7-4698-991b-7ee674b11079",
                 "bf0a6d94-58f0-4ad2-bdf2-c4a31c7dd856",
-                "10f0d9bc-5c9e-4216-9a94-f3a2ff53bb5f"
+                "6a2d0734-bd9e-4ddf-946e-caa52498cb75"
```
Comment (star-e) [inline cc.config.json:208]: Change `post-final` to `tone-mapping`.

File: editor/assets/default_renderpipeline/builtin-pipeline.ts [hunk 2]
```diff
@@ -178,9 +188,17 @@ function setupCameraConfigs(
 
     cameraConfigs.useFullPipeline = (camera.visibility & (Layers.Enum.DEFAULT)) !== 0;
 
-    cameraConfigs.enableShadowMap = camera.scene
-        ? camera.scene.mainLight !== null && camera.scene.mainLight.shadowEnabled
-        : false;
+    cameraConfigs.enableMainLightShadowMap = pipelineConfigs.shadowEnabled
+        && !pipelineConfigs.usePlanarShadow
+        && camera.scene !== null
+        && camera.scene.mainLight !== null
```
Comment (dumganhar) [inline editor/assets/default_renderpipeline/builtin-pipeline.ts]: Would `camera.scene` or `camera.scene.mainLight` be `undefined`?
I found that we just check whether they're equal to `null` but did not consider `undefined`.
Maybe just use `&& camera.scene && camera.scene.mainLight && camera.scene.mainLight.shadowEnabled` which also could reduce some bytes.

Comment (star-e) [reply editor/assets/default_renderpipeline/builtin-pipeline.ts]: The type is `T | null`, so I just checked `null`. I will make the change.

Comment (star-e) [reply editor/assets/default_renderpipeline/builtin-pipeline.ts]: Changed.

**Comment by star-e** [comment]
@cocos-robot run test cases

**Review by dumganhar — APPROVED** [review]

**Pull request merged**